packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -1517,9 +1517,9 @@ describe('<Autocomplete />', () => {
 
       checkHighlightIs(listbox, 'two');
 
-      // three option is added and autocomplete re-renders, reset the highlight
+      // three option is added and autocomplete re-renders, do not reset the highlight
       setProps({ options: ['one', 'two', 'three'] });
-      checkHighlightIs(listbox, null);
+      checkHighlightIs(listbox, 'two');
```

Comment:
It's not clear to me if this is not a regression. Some might expect that new options = a reset of the highlight.

Reply:
Are you suggesting that we allow the highlight index to reset, but prevent scroll to top?

Reply:
@oliviertassinari seeing the number of issues reported about this, the community seems to have a different view on this. It may be confusing to have highlight gone when data is loaded on demand.

@cjtroiano I'm wondering about another scenario that your PR doesn't cover - what if the number of items decreased, but the highlighted item is still among them. IMO we should keep the highlight there as well:

```jsx
it('should keep focus on selected option and not reset to top option when options updated', () => {
  const { setProps } = render(
    <Autocomplete
      open
      options={['one', 'two', 'three']}
      renderInput={(params) => <TextField {...params} autoFocus />}
    />,
  );
  const textbox = screen.getByRole('combobox');
  const listbox = screen.getByRole('listbox');

  fireEvent.keyDown(textbox, { key: 'ArrowDown' }); // goes to 'one'
  fireEvent.keyDown(textbox, { key: 'ArrowDown' }); // goes to 'two'

  checkHighlightIs(listbox, 'two');

  // three option is added and autocomplete re-renders, do not reset the highlight
  setProps({ options: ['two', 'three'] });
  checkHighlightIs(listbox, 'two');
});
```